### PR TITLE
Don't crash when building AI Chat context menu on PWA windows

### PR DESCRIPTION
**Pull request by petemill**
Resolves https://github.com/brave/brave-browser/issues/36995

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: chromium_src/chrome/browser/renderer_context_menu/render_view_context_menu.cc [hunk 1]
```diff
@@ -385,11 +385,13 @@ void BraveRenderViewContextMenu::CopyTextFromImage() {
...
 #if BUILDFLAG(ENABLE_AI_CHAT)
 bool BraveRenderViewContextMenu::IsAIChatEnabled() const {
-  return ai_chat::IsAIChatEnabled(GetProfile()->GetPrefs()) &&
+  return !params_.selection_text.empty() &&
+         ai_chat::IsAIChatEnabled(GetProfile()->GetPrefs()) &&
          brave::IsRegularProfile(GetProfile()) &&
          GetProfile()->GetPrefs()->GetBoolean(
              ai_chat::prefs::kBraveAIChatContextMenuEnabled) &&
-         !params_.selection_text.empty();
+         ai_chat::AIChatTabHelper::FromWebContents(source_web_contents_) &&
```
Comment (petemill) [inline chromium_src/chrome/browser/renderer_context_menu/render_view_context_menu.cc:393]: For now (but not forever), if we're not creating a Tab Helper for this WebContents, then we're not enabling Leo in this context, so don't show those items. It's guarded on the same `IsRegularProfile` and `ai_chat::IsAIChatEnabled` that this function is already checking, but seems like we don't need to duplicate that check at the moment.

File: chromium_src/chrome/browser/renderer_context_menu/render_view_context_menu.cc [hunk 2]
```diff
@@ -385,11 +385,13 @@ void BraveRenderViewContextMenu::CopyTextFromImage() {
...
+         !IsInProgressiveWebApp();
```
Comment (petemill) [inline chromium_src/chrome/browser/renderer_context_menu/render_view_context_menu.cc:394]: This is the fix for the crash

**Review by emerick — APPROVED** [review]
LGTM

File: chromium_src/chrome/browser/renderer_context_menu/render_view_context_menu.cc (hunk 2 above)
Comment (bsclifton) [reply chromium_src/chrome/browser/renderer_context_menu/render_view_context_menu.cc:393]: Do we still need the other check?
`!params_.selection_text.empty()`

ex: should it be:
```
!params_.selection_text.empty() &&
!IsInProgressiveWebApp();
```

**Review by bsclifton — APPROVED** [review]
Change LGTM! 😄

**Comment by bsclifton** [comment]
Hmm - seeing this on Android run
```
 Running Python 3 presubmit upload checks ...
   checking for commit objects in tree took a long time: 11.9s
   11.9s to run ChecksCommon from /home/ubuntu/workspace/pr-brave-browser-fix-crash-pwa-leo-menu-android/src/brave/PRESUBMIT.py.
 ** Presubmit ERRORS: 1 **
 You added one or more #includes that violate checkdeps rules.
   brave/browser/renderer_context_menu/test/render_view_context_menu_unittest.cc
       Illegal include: "gtest/gtest.h"
     Because of no rule applying.
 ```

**Comment by LaurenWags** [comment]
Verification in progress with

```
Brave	1.66.32 Chromium: 123.0.6312.58 (Official Build) nightly (x86_64) 
Revision	05c063551294bf8fc9c65980112a22883ba52b82
OS	macOS Version 13.6.4 (Build 22G513)
```

**Pull request merged**